rtlib/src/SepDisegni.cpp

```diff
@@ -0,0 +1,275 @@
+    // BUT a 8bits unsigned int (the one we are working with) can contain values from 0 to 255
+    // so the possible negative number will be truncated
+    Mat imgLaplacian;
+    filter2D(src, imgLaplacian, CV_32F, kernel);
+    Mat sharp;
+    src.convertTo(sharp, CV_32F);
+    Mat imgResult = sharp - imgLaplacian;
+    // convert back to 8bits gray scale
+    imgResult.convertTo(imgResult, CV_8UC3);
+    imgLaplacian.convertTo(imgLaplacian, CV_8UC3);
+    // imshow( "Laplace Filtered Image", imgLaplacian );
+    namedWindow("New Sharped Image", 0);
+    imshow( "New Sharped Image", imgResult );
+    waitKey(0);
+
+
+    //Image Median Blur
+
+    //Median Blur Image
+    medianBlur(imgResult, imgResult, 7);
+    namedWindow("Blurred Image", 0);
+    imshow("Blurred Image", imgResult);
+    waitKey(0);
+
+
+    // Create binary image from source image
+    Mat bw;
+    cvtColor(imgResult, bw, COLOR_BGR2GRAY);
+    threshold(bw, bw, 40, 255, THRESH_BINARY | THRESH_OTSU);
+    namedWindow("Binary Image", 0);
+    imshow("Binary Image", bw);
+    waitKey(0);
+
+
+
+
+
+    // Perform the distance transform algorithm
+    Mat dist;
+    distanceTransform(bw, dist, DIST_L2, 3);
+    // Normalize the distance image for range = {0.0, 1.0}
+    // so we can visualize and threshold it
+    normalize(dist, dist, 0, 1.0, NORM_MINMAX);
+    namedWindow("Distance Transform Image", 0);
+    imshow("Distance Transform Image", dist);
+    waitKey(0);
+
+
+
+    // Threshold to obtain the peaks
+    // This will be the markers for the foreground objects
+    threshold(dist, dist, 0.4, 1.0, THRESH_BINARY);
+    // Dilate a bit the dist image
+    Mat kernel1 = Mat::ones(3, 3, CV_8U);
+    dilate(dist, dist, kernel1);
+    namedWindow("Peaks", 0);
+    imshow("Peaks", dist);
+    waitKey(0);
+
+
+
+    // Create the CV_8U version of the distance image
+    // It is needed for findContours()
+    Mat dist_8u;
+    dist.convertTo(dist_8u, CV_8U);
+    // Find total markers
+    vector<vector<Point> > contours;
+    findContours(dist_8u, contours, RETR_EXTERNAL, CHAIN_APPROX_SIMPLE);
+    // Create the marker image for the watershed algorithm
+    Mat markers = Mat::zeros(dist.size(), CV_32S);
+    // Draw the foreground markers
+    for (size_t i = 0; i < contours.size(); i++) {
+        drawContours(markers, contours, static_cast<int>(i), Scalar(static_cast<int>(i) + 1), -1);
+    }
+    // Draw the background marker
+    circle(markers, Point(5, 5), 3, Scalar(255), -1);
+    markers.convertTo(markers, CV_8U);
+    namedWindow("Markers", 0);
+    imshow("Markers", markers * 10000);//FIX ITTT convert to correct C_16F
+    waitKey(0);
+    markers.convertTo(markers, CV_32S);
+
+
+
+    // Perform the watershed algorithm
+    watershed(imgResult, markers);
+    Mat mark;
+    markers.convertTo(mark, CV_8U);
+    bitwise_not(mark, mark);
+    //    imshow("Markers_v2", mark); // uncomment this if you want to see how the mark
+    // image looks like at that point
```

Comment:
What's the point of this?

Reply:
It comments I did not see when I copied over the source code I deleted it